### PR TITLE
add topic partitions to style guide

### PR DESCRIPTION
**Pull request by Deflaimun**

### CODE REVIEW

File: meta-docs/STYLE-GUIDE.adoc [hunk 1]
```diff
@@ -1100,6 +1100,9 @@ Incorrect: To enable Shadow Indexing on a cluster, set the following parameters
 |topic
 |A topic is a stream of related events. A doc topic is content in Redpanda product doc library.
 |
+|topic partition
+|Write without hyphens. A topic may be partitioned through multiple brokers. A "topic partition" represents this logical separation on Redpanda. It is managed by Raft, applying voting mechanisms such as Round Robin. In contrast, other Kafka applications use ZooKeeper to manage their partitions. In Redpanda this is managed natively by Raft.
```
Comment (micheleRP) [inline meta-docs/STYLE-GUIDE.adoc:1104]:
```suggestion
|Write as 2 words, not 1 hyphenated word. 
```

Comment (micheleRP) [reply meta-docs/STYLE-GUIDE.adoc:1104]: @paulo please also fix the formatting of the word list table while you're in this file!

Comment (Deflaimun) [reply meta-docs/STYLE-GUIDE.adoc]: Fixed. see https://github.com/redpanda-data/docs-site/blob/style-guide/meta-docs/STYLE-GUIDE.adoc

**Review by micheleRP — APPROVED** [review]
nice fix, thanks Paulo!

**Comment by Deflaimun** [comment]
Wrote the definition here in the glossary entry 
https://github.com/redpanda-data/docs/pull/909/files

**Pull request merged**